packages/transport/src/modules/tokens/TokenController.ts

```diff
@@ -186,12 +189,16 @@ export class TokenController extends TransportController {
         return await this.loadPeerTokenByReference(reference, ephemeral);
     }
 
-    public async loadPeerTokenByReference(tokenReference: TokenReference, ephemeral: boolean): Promise<Token> {
-        return await this.loadPeerToken(tokenReference.id, tokenReference.key, ephemeral);
+    private async loadPeerTokenByReference(tokenReference: TokenReference, ephemeral: boolean): Promise<Token> {
+        return await this.loadPeerToken(tokenReference.id, tokenReference.key, ephemeral, tokenReference.forIdentityTruncated);
     }
 
-    public async loadPeerToken(id: CoreId, secretKey: CryptoSecretKey, ephemeral: boolean): Promise<Token> {
+    private async loadPeerToken(id: CoreId, secretKey: CryptoSecretKey, ephemeral: boolean, forIdentityTruncated?: string): Promise<Token> {
         const tokenDoc = await this.tokens.read(id.toString());
+        if (!tokenDoc && forIdentityTruncated && !this.parent.identity.address.toString().endsWith(forIdentityTruncated)) {
```

Comment:
why do you even query the db?

Reply:
the idea was that the personalization check will only be executed when loading the token for the first time (if it's not ephemeral)

Reply:
I would love to see this check at the first place and 

1. check if you created the token or
2. you can access the token

It's so hard to process why you have to query the DB before doing that check.

Reply:
How do you want to check if you created the token without querying the DB? Not allowing you to load your own token would not be that bad but I'd like to do without it. Coding it as `const isOwnTokenOrAlreadyLoadedToken = !!tokenDoc` would also be a way to clean up

Reply:
I think I understand the issue now. 

We do not have access to the creator of the token if we didn't fetch the token. But we can skip the "is the token mine" step when the token is already in the DB. In this case I'm ok with this check.